org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -782,4 +786,180 @@ private String convertDebuggerSourceToClient(String uri) {
     private Types.Message convertDebuggerMessageToClient(String message) {
         return new Types.Message(this.messageId.getAndIncrement(), message);
     }
+
+    private class VariableHandler {
+        private boolean showStaticVariables = true;
+        private IVariableFormatter variableFormatter;
+        private RecyclableObjectPool<ThreadReference, Object> objectPool;
+        private Map<String, Object> options;
+        
+        public VariableHandler(IVariableFormatter variableFormatter, boolean showStaticVariables,
```

Comment:
showStaticVariables should be an option that is passed to the formatters from the variable request.